tools/engine_tool/lib/src/logger.dart

```diff
@@ -106,6 +111,17 @@ class Logger {
     _logger.level = l;
   }
 
+  /// Record a log message level [Logger.error] and throw a FatalError.
+  void fatal(
+    Object? message, {
+    int indent = 0,
+    bool newline = true,
+    bool fit = false,
+  }) {
+    _emitLog(errorLevel, message, indent, newline, fit);
```

Comment:
Just FYI. There was a discussion a while back about how to exit from a command line program with @matanlurey and others. This is an example of why calling `dart:io`'s `exit()` will have unexpected results. This `_emitLog()` writes to `io.stderr` which is an `IOSink`. It does buffered writes. The line below this synchronously throws an exception, which is caught in `main()`. If `main()` caught it and synchronously called `dart:io` `exit()`, then the process might end before the log message is flushed to stderr.

Reply:
We don't call exit() from main. We set the exitCode and let main return as normal. Do we need to add a flush somewhere?

Reply:
No we're all good. Just wanted to relate this as an example to a discussion we were having earlier in chat.

Reply:
Thanks!